### PR TITLE
remove default request timeouts, obey custom timeouts via prefer header

### PR DESCRIPTION
**Pull request by irshadaj**
## Description
* Remove the default timeouts of 30 seconds for all API requests
* Continue to support the `prefer` header to specify custom timeouts for individual requests

## How Has This Been Tested?

* Updated unit tests
* Manual testing to verify that the prefer header is processed as supplied

Hitting any endpoint without a `Prefer` request header returns a response without the `Preference-Applied` header in the response:
no wait

Adding the `Prefer` header to the request, we see the `Preference-Applied` header in the response:
wait

## Types of changes

-   [ ] Chore (a change that does not modify the application functionality)
-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [ ] Documentation updates are needed, and have been made accordingly.
-   [x] I have added and/or updated tests to cover my changes.
-   [x] All new and existing tests passed.
-   [ ] My changes include a database migration.

### CODE REVIEW

File: cmd/api/src/api/middleware/middleware.go [hunk 1]
```diff
@@ -195,21 +199,16 @@ func ParseHeaderValues(values string) map[string]string {
...
-	if hasWaitSpecified {
+func parsePreferHeaderWait(value string) (time.Duration, error) {
+	if waitStr, hasWaitSpecified := ParseHeaderValues(value)["wait"]; hasWaitSpecified {
 		if parsedNumSeconds, err := strconv.Atoi(waitStr); err != nil {
 			return 0, err
 		} else {
 			return time.Second * time.Duration(parsedNumSeconds), nil
 		}
+	} else {
+		return time.Duration(0), nil
```
Comment (mistahj67) [inline cmd/api/src/api/middleware/middleware.go:210]: I don't think this assertion is needed 🤷

File: cmd/api/src/api/v2/version.go
Comment (mistahj67) [inline cmd/api/src/api/v2/version.go:1]: are these changes intended?

File: cmd/api/src/queries/graph_test.go [hunk 2]
```diff
@@ -97,21 +93,21 @@ func TestGraphQuery_RawCypherSearch(t *testing.T) {
...
+		require.Equal(t, time.Minute*10, config.Timeout)
 
 		return nil
 	}).Times(2)
 
 	// Unset the user-set timeout in the BH context to validate QC runtime reduction of a complex query
-	outerBHCtxInst.Timeout.UserSet = false
-	outerBHCtxInst.Timeout.Value = time.Minute
+	// This will be set to a default of 30 min, with a reduction factor of 3, so we should have a 10 min config timeout
+	outerBHCtxInst.Timeout = 0
```
Comment (mistahj67) [inline cmd/api/src/queries/graph_test.go:103]: I'm confused here, the comment says it should be 30 minutes but you have 0 here

File: cmd/api/src/api/middleware/middleware.go (hunk 1 above)
Comment (mistahj67) [reply cmd/api/src/api/middleware/middleware.go:210]: Was referring to the type assertion `time.Duration(0)` vs just 0

**Comment by rvazarkar** [comment]
Tests look good and nothing stands out logic wise that would be a problem. Are there existing tests to validate that our timeouts actually work as intended? Maybe something that just sleeps for 10 seconds with a 5 second wait header to ensure we get a timeout
